### DISCUSSION TITLE
exterior_facet integral cannot be used with TensorProductCell

### DISCUSSION THREAD

**Discussion by cjham** [firedrakeproject/firedrake#3858] [Firedrake support]
I am having difficulty implementing a Neuman boundary condition. The boundary condition I want to use is made up of forming a vector, using as_vector(), which I then cross with the normal vector and then integrate over the surface. The error that comes up "exterior_facet integral cannot be used with TensorProductCell" seems to suggest that this is not allowed. Is that correct? I guess I could calculate the cross product by hand using the components of the normal n[0] etc and use that instead is that the best way to go? Thank you for your help.

**Comment by colinjcotter** [comment]
can you show us the code?

**Comment by colinjcotter** [comment]
This is because an extruded mesh was being used, which requires ds_t, ds_b etc instead of ds.
